components/org.wso2.carbon.identity.application.authenticator.adapter/src/main/java/org/wso2/carbon/identity/application/authenticator/adapter/AuthenticationRequestBuilder.java

```diff
@@ -0,0 +1,158 @@
+        actionRequestBuilder.flowId((String) eventContext.get(AuthenticatorAdapterConstants.FLOW_ID));
+        actionRequestBuilder.actionType(getSupportedActionType());
+        actionRequestBuilder.event(getEvent(request, context));
+        AllowedOperationBuilder allowedOperationBuilder = new AllowedOperationBuilder();
+        actionRequestBuilder.allowedOperations(allowedOperationBuilder.getAllowedOperations());
+
+        return actionRequestBuilder.build();
+    }
+
+    private Event getEvent(HttpServletRequest request, AuthenticationContext context)
+            throws ActionExecutionRequestBuilderException {
+
+        AuthenticatedUser lastAuthenticatedUser = context.getLastAuthenticatedUser();
+        String tenantDomain = context.getTenantDomain();
+
+        AuthenticationRequestEvent.Builder eventBuilder = new AuthenticationRequestEvent.Builder();
+        eventBuilder.tenant(new Tenant(String.valueOf(IdentityTenantUtil.getTenantId(tenantDomain)), tenantDomain));
+        if (lastAuthenticatedUser != null) {
+            eventBuilder.user(getUserForEventBuilder(lastAuthenticatedUser));
+            eventBuilder.organization(getOrganizationForEventBuilder(lastAuthenticatedUser));
+            eventBuilder.userStore(new UserStore(lastAuthenticatedUser.getUserStoreDomain()));
+        }
+        eventBuilder.currentStepIndex(context.getCurrentStep());
+        eventBuilder.authenticatedSteps(getAuthenticatedStepsForEventBuilder(context));
+        eventBuilder.request(getRequest(request));
+        return eventBuilder.build();
+    }
+
+    private User getUserForEventBuilder(AuthenticatedUser authenticatedUser)
+            throws ActionExecutionRequestBuilderException {
+
+        try {
+            return new AuthenticationRequestUser(authenticatedUser.getUserId(), authenticatedUser);
+        } catch (UserIdNotFoundException e) {
+            throw new ActionExecutionRequestBuilderException("User ID not found for current authenticated user.", e);
+        }
+    }
+
+    private Organization getOrganizationForEventBuilder(AuthenticatedUser authenticatedUser) throws ActionExecutionRequestBuilderException {
+
+        try {
+            String organizationId = authenticatedUser.getUserResidentOrganization();
+            if (organizationId != null && !organizationId.isEmpty()) {
+                String organizationName = AuthenticatorAdapterDataHolder.getInstance().getOrganizationManager()
+                        .getOrganizationNameById(authenticatedUser.getUserResidentOrganization());
+                return new Organization(authenticatedUser.getUserResidentOrganization(), organizationName);
+            }
+        } catch (OrganizationManagementException e) {
+            throw new ActionExecutionRequestBuilderException("Error occurred while retrieving organization name " +
+                    "of the authorized user", e);
+        }
+        return null;
```

Comment:
If organization is not present still the request may be initiated. At this point organization is not a mandatory property

Reply:
Fixed